apps/pipelines/nodes/base.py

```diff
@@ -182,6 +182,8 @@ def process_conditional(self, state: PipelineState, node_id: str | None = None)
         output_map = self.get_output_map()
         output_handle = next((k for k, v in output_map.items() if v == conditional_branch), None)
         state["outputs"][node_id]["output_handle"] = output_handle
+        if self.tag_output_message:
+            state["output_message_metadata"].setdefault("tags", []).append(conditional_branch)
```

Comment:
I think we should use a separate data structure for tags and metadata. Also the tag name should include the node name:

```suggestion
            state["output_message_tags"].setdefault("tags", []).append(f"{self.name}:{conditional_branch}")
```